### PR TITLE
Add Bookmarking (Local to device)

### PR DESCRIPTION
**Pull request by joelklabo**
| Example |
| - |
| Simulator Screen Recording - iPhone 14 Pro - 2023-02-20 at 21 29 40 |

**UPDATED**
No settings button anymore, moved to navigation bar in bookmarks view

| Example (Dark) | Empty State | Empty State (Dark) | Settings | 
|      -          |      -         |      -      |       -      |
| example-dark | empty | example-dark | settings |

This just saves them (as `JSON`) to user defaults so only local to device.

### CODE REVIEW

**Comment by jb55** [comment]
dope!

File: damus/Models/BookmarksManager.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+//
+
+import Foundation
+
+class BookmarksManager {
+    static private let userDefaults = UserDefaults.standard
+    
+    static var bookmarks: [String] {
+        get {
+            return userDefaults.stringArray(forKey: "bookmarks") ?? []
```
Comment (jb55) [inline damus/Models/BookmarksManager.swift:15]: this needs to be keyed on pubkey

Comment (jb55) [reply damus/Models/BookmarksManager.swift:15]: I have a pretty dumb helper for that: pk_setting_key

**Review by jb55 — COMMENTED** [review]
looks good, just one change needed

File: damus/Models/BookmarksManager.swift (hunk 1 above)
Comment (joelklabo) [reply damus/Models/BookmarksManager.swift:15]: Oh, right, multiple accounts? I can update that 🤙

**Comment by joelklabo** [comment]
Ok, updated @jb55

**Comment by jb55** [comment]
looks great! testing

**Comment by jb55** [comment]
just need a way to delete and then I can merge

**Comment by joelklabo** [comment]
> just need a way to delete and then I can merge

Must be a bug but it should show "Un-bookmark" if you've already bookmarked it. I'll check again, would that work? If it worked?

**Comment by jb55** [comment]
It shows "Bookmark".

**Comment by jb55** [comment]
Also we probably shouldn't have "clear bookmarks" in settings, we can just add it as a button on the top-right of the toolbar on the bookmarks view

**Comment by joelklabo** [comment]
> Also we probably shouldn't have "clear bookmarks" in settings, we can just add it as a button on the top-right of the toolbar on the bookmarks view

Yeah, that would be better. I'll try that

**Comment by joelklabo** [comment]
Ok, I think this is all fixed up now, updated the GIF in the description with current behavior @jb55

**Comment by joelklabo** [comment]
Looks like one more thing to figure out after rebasing with master:

`Cannot convert value of type 'Binding<[NostrEvent]>' to expected argument type 'EventHolder'`

**Comment by joelklabo** [comment]
> Looks like one more thing to figure out after rebasing with master:
> 
> `Cannot convert value of type 'Binding<[NostrEvent]>' to expected argument type 'EventHolder'`

Should be fixed now

**Pull request closed without merging**